src/software/ai/navigator/navigator.h

```diff
@@ -1,42 +1,154 @@
 #pragma once
 
 #include "software/ai/intent/intent.h"
+#include "software/ai/intent/intent_visitor.h"
+#include "software/ai/navigator/navigator.h"
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/path_planner/straight_line_path_planner.h"
+#include "software/ai/navigator/path_planner/theta_star_path_planner.h"
 #include "software/ai/primitive/primitive.h"
+#include "software/util/parameter/dynamic_parameters.h"
 #include "software/world/world.h"
 
 /**
- * An abstraction for all navigation operations performed by our AI. The navigator is
- * responsible for all navigation, path-planning, and collision-avoidance for our robots.
- *
- * This is an Abstract, pure-virtual class. It is meant to define the interface that all
- * Navigator modules must follow. Other classes should inherit from this class and
- * implement the methods to create a useable Navigator class. This allows us to
- * potentially have multiple different navigators that we can swap out or combine at
- * runtime.
+ * This Navigator is a simple navigator that will convert the given Intents
+ * into their respective Primitives, without doing any intelligent navigation. It is meant
+ * to provide an example of how to implement a navigator and act as a starting point for
+ * future navigators to build off of.
```

Comment:
Is this description correct still? Since this is the only navigator now?

Reply:
I will update

Reply:
actually I want to say will take a path planner and intents and return primitives, but that won't be accurate until next PR. Can I leave this for now?

Reply:
I mean, should we remove the bits about "not doing intelligent navigation" and "provide an example"? If this is the only navigator now then those certainly aren't accurate right?

Reply:
I agree the comment isn't accurate, but are you ok with leaving this for the next PR where path planner is passed into navigator?

Reply:
My hand has been forced by a merge conflict. I will update and this probably won't get merged today :(